blockchain/pallets/minting/src/lib.rs

```diff
@@ -150,46 +107,37 @@ pub mod pallet {
 
     #[pallet::error]
     pub enum Error<T> {
-        InvalidIdentitySignature,
-        MismatchedFractalIdentity,
-        NonIncreasingNonce,
         NoIdentityRegistered,
         ExtensionDoesNotExtendExistingDataset,
         MustSpecifyFractalIdWithMultipleIds,
         FractalIdNotRegisteredToAccount,
+        MustBeFractal,
     }
 
     #[pallet::call]
     impl<T: Config> Pallet<T> {
-        /// Register the provided Fractal ID to the origin account.
-        #[pallet::weight(10_000 + T::DbWeight::get().reads_writes(2, 2))]
+        #[pallet::weight(0)]
```

Comment:
Out of curiosity: could this be computed in terms of who the origin is?

Reply:
Yes. By returning DispatchResultWithPostInfo I think. https://docs.rs/frame-support/2.0.0-alpha.6/frame_support/dispatch/type.DispatchResultWithPostInfo.html